java-version/accelerator.yaml

```diff
@@ -27,10 +27,34 @@ engine:
         regex:
           pattern: "sourceCompatibility = .*"
           with: "'sourceCompatibility = ''' + #javaVersion + ''''"
-    - include: [ "config/workload.yaml" ]
+    - include: [ "build.gradle.kts" ]
       chain:
       - type: ReplaceText
-        condition: "#javaVersion == '17'"
-        substitutions:
-          - text: "spec:"
-            with: "'spec:\n  build:\n    env:\n    - name: BP_JVM_VERSION\n      value: \"17\"'"
+        regex:
+          pattern: "(?<unmodified>JavaVersion\\.VERSION_)(\\d*)"
+          with: "'${unmodified}' + #javaVersion.replace('.', '_')"
+    - condition: "#files.contentsOf('config/workload.yaml').get().contains('- name: BP_JVM_VERSION')"
```

Comment:
This is not what `#files` has been introduced for ... and it shows...

Reply:
Currently there is no other way to provide two different transformation depending on the fact whether BP_JVM_VERSION is already provided in the host accelerator workload.yaml.
Even though the solution is not perfect, it is still better than the one we had before. Let's move on with this one and create a ticket to provide a proper solution.